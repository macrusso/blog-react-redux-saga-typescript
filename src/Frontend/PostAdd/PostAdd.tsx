import React from 'react';
import { Formik } from 'formik';
import { IPostPartial } from 'src/Entities';

interface IPostAddProps {
  addPost: (post: IPostPartial) => void;
}

const PostAdd: React.SFC<IPostAddProps> = props => {
  const { addPost } = props;
  return (
    <div>
      <h1>Anywhere in your app!</h1>
      <Formik
        initialValues={{ title: '', body: '' }}
        onSubmit={values => {
          addPost({ ...values, userId: 1 });
        }}
      >
        {({ values, handleChange, handleBlur, handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <input
              type="title"
              name="title"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.title}
            />
            <input
              type="body"
              name="body"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.body}
            />
            <button type="submit">Submit</button>
          </form>
        )}
      </Formik>
    </div>
  );
};

export default PostAdd;
