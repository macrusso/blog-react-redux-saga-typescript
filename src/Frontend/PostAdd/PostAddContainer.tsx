import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { IAction, IStoreState } from '../../types';
import { ErrorBoundary } from '../Shared';
import { postSelectors, IPostPartial, postActions } from '../../Entities';
import { PostAdd } from '.';
import { appSelectors } from '../../App';

type IPostAddContainerProps = IStateToProps & IDispatchToProps;

class PostAddContainer extends Component<IPostAddContainerProps> {
  public render() {
    const props = this.props;

    return (
      <ErrorBoundary>
        <PostAdd {...props} />
      </ErrorBoundary>
    );
  }
}

interface IStateToProps {
  error?: string;
  userId: string;
}

interface IDispatchToProps {
  addPost: (post: IPostPartial) => void;
}

const mapStateToProps = (state: IStoreState) => ({
  error: postSelectors.getError(state),
  userId: appSelectors.getUserId(state),
});

const mapDispatchToProps = (dispatch: Dispatch<IAction>) => ({
  addPost: (post: IPostPartial) => {
    dispatch(postActions.addPostRequest(post));
  },
});

export default connect<IStateToProps, IDispatchToProps, any>(
  mapStateToProps,
  mapDispatchToProps
)(PostAddContainer);
