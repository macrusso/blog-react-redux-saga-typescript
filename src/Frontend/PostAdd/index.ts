import PostAddContainer from "./PostAddContainer";
import PostAdd from "./PostAdd";

export { PostAddContainer, PostAdd };
