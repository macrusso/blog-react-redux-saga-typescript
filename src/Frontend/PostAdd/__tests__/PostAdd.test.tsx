import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import PostAdd from '../PostAdd';

describe('PostAdd', () => {
  test('Snapshot matches', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <PostAdd addPost={jest.fn()} />
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
