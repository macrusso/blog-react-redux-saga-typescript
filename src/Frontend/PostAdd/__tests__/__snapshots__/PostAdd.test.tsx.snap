// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostAdd Snapshot matches 1`] = `ShallowWrapper {}`;
